### ISSUE TITLE
Any way to only cache the vcpkg executable and tools?

### ISSUE THREAD

**Issue by ekilmer** [lukka/run-vcpkg#49]
Hello, thank you for this great action!

I have maybe a weird request: Is there any way to cache only the vcpkg executable and tools that are downloaded (or to put another way: exclude the build artifacts of the libraries)?

I'm working on some projects that have dependencies on some very large projects, like LLVM, and the cache grows well beyond GitHub Actions' allowable 5GB limit.

As an alternative, I'm exploring the NuGet binarycaching that will hopefully be able to pull the pre-built libraries that are compatible (read: matching hashes) with the build environment and vcpkg checkout to workaround these limits (... I still need to test if there are NuGet upload size limits). It would be nice to take advantage of the GitHub Action caches for vcpkg binary and the tools (ninja, cmake, etc.) that are downloaded though.

**Related**: I can't seem to find where the `vcpkgArtifactIgnoreEntries` key is used in the implementation of this action. It looks like the artifact entries are hardcoded here. Am I missing something? https://github.com/lukka/run-vcpkg/blob/fc243fce3040cd0f085c9704118f849ebacbe9d3/src/vcpkg-action.ts#L37-L42

**Comment by lukka** [comment]
@ekilmer Thanks for the feedback! To achieve caching of the built vcpkg and related tools (but not of installed packages), just use `setupOnly:true`, e.g.:
        
```yaml
    uses: lukka/run-vcpkg@v4
        with:
          setupOnly: true
          vcpkgDirectory: '${{ runner.workspace }}/b/vcpkg'
          vcpkgGitCommitId: '${{ matrix.vcpkgCommitId}}'
          # Ensure the cache key changes any time the content of the response file changes.
          appendedCacheKey: ${{ hashFiles( env.VCPKGRESPONSEFILE ) }}
```
which i copied from the [sample](https://github.com/lukka/CppBuildTasks-Validation/actions/runs/339051088/workflow).

About `vcpkgArtifactIgnoreEntries` input, I've created it but it is not being used on GitHub action. It being used in the [CppBuildTasks Azure Devops tasks](https://marketplace.visualstudio.com/items?itemName=lucappa.cmake-ninja-vcpkg-tasks). I may revisit its doc and implementation.

[Here](https://github.com/lukka/run-cmake-vcpkg-action-libs/blob/12e63395af1a28c091143e7e55349b570eeb8833/packages/run-vcpkg-lib/src/vcpkg-runner.ts#L48) there is its usage (in a npm package)

**Comment by ekilmer** [comment]
Thank you for the quick response and the links! They are very helpful 🙂 

Honestly, I've been having trouble debugging exactly what's happening on the CI runners, given that LLVM is a huge project with high memory and space requirements, but I think I know what's been happening.

**First**, LLVM with debug symbols is _huge_:
```
$ du -hs buildtrees/llvm
42G     buildtrees/llvm

$ du -hs packages/llvm_x64-linux/
12G     packages/llvm_x64-linux/
```
And the CI runners for Actions have anywhere [from 10-14GB of storage space](https://github.community/t/bigger-github-hosted-runners-disk-space/17267/9), so this has been giving me weird failure results on Linux runners. Oddly, a `macos` runner is able to build a debug version of LLVM, but this might be a bug with not correctly calculating disk space on those machines 🤷 

**Second**, if a runner does not immediately fail after using vcpkg to install a huge package (like with a `vcpkg install .... || true`, bad practice, I know...), this action will still try to cache the `vcpkg/installed` directory, which, even compressed, could still be larger than the 5GB max and fail to cache anything at all, including the `vcpkg` executable and downloaded tools.

-----

I've created a test repo for this issue here https://github.com/ekilmer/test-run-vcpkg and ran two Actions to show that the `installed` directory is being cached even with `setupOnly: true`.

Here is the first run where everything is built from scratch due to no cache: https://github.com/ekilmer/test-run-vcpkg/runs/1336245863?check_suite_focus=true#step:4:101

And here is where `vcpkg` executable is not built again (expected), but `fmt` is also not built because the `installed` directory is cached: https://github.com/ekilmer/test-run-vcpkg/runs/1336373577?check_suite_focus=true#step:4:21

Caching the `installed` directory makes a lot of sense, so I'm not sure if this explanation/scenario is convincing enough to warrant any changes to this project, but I just thought I'd point it out and listen to what you have to say.

Thank you again!

**Comment by lukka** [comment]
@ekilmer you are absolutely right, the `installed` directory is being cached even when using `setupOnly:true`. 
Indeed the `run-vcpkg` schedules a post-step that will be run at the very end of the workflow and it will store data in cache. 
An easy workaround it to rename the folders you do not want to cache, e.g.:

```yaml
   run: |
      mv $VCPKG_ROOT/installed $VCPKG_ROOT/installed_moved_away
   shell: bash
```

_Note:_ Recently I modified the [get-cmake](https://github.com/marketplace/actions/get-cmake) action to cache CMake and ninja right away after their installation (and not relying on the post-step), and I thought mistakenly that I did the same for`run-vcpkg`: in fact the `run-vcpkg` is still relying on the post-step to cache data, and i plan to solve [issue 24](https://github.com/lukka/run-vcpkg/issues/24) sooner or later.
One difficulty is that when doing this change, I need to carefully think about workflows that are:
   - relying on automatic caching.
   - using `setupOnly:true`.

The change would mean for those workflows to not cache the `installed` directory anymore. This could be solved with a new major version (very likely), or a new `input` (not likely).
